Dockerfile

```diff
@@ -0,0 +1,33 @@
+# Python 3.10 w/ Nvidia Cuda
+FROM nvidia/cuda:11.8.0-devel-ubuntu22.04 AS env_base
+
+# Install Pre-reqs
+RUN apt-get update && apt-get install --no-install-recommends -y \
+    git vim nano build-essential python3-dev python3-venv python3-pip gcc g++ ffmpeg
+
+# Setup venv
+RUN pip3 install virtualenv
+RUN virtualenv /venv
+ENV VIRTUAL_ENV=/venv
+RUN python3 -m venv $VIRTUAL_ENV
+ENV PATH="$VIRTUAL_ENV/bin:$PATH"
+RUN pip3 install --upgrade pip setuptools && \
+    pip3 install torch torchvision torchaudio
+
+# Set working directory
+WORKDIR /app
+
+# Clone the repo
+RUN git clone https://github.com/rsxdalv/tts-generation-webui.git
+
+# Set working directory to the cloned repo
+WORKDIR /app/tts-generation-webui
+
+# Install all requirements
+RUN pip3 install -r requirements.txt
+RUN pip3 install -r requirements_audiocraft.txt
```

Comment:
There's a script update.py which does the extra installs.
There are downsides to using it, so I'm wondering if this is an optimization for docker build or if there's a bug with update.py.

Reply:
The update.py approach is a bit unique. I've never seen Python used to manage installs via cmd. It seems a bit brittle. 

The Docker RUN commands would ideally get collapsed into one which would aid in better caching of layers within the image.

Reply:
It makes sense. I wanted to have a single requirements.txt but there are (incorrectly) incompatible requirements in the downstream packages. And also there's plenty of issues for different users that prevent it from being done. I think for now this is a solid approach, also given that:
1. a docker image should be one that _builds_ not _partially fails_
2. it's much more clear rather than a custom update script with error handling.

Lastly I wanted to add that for tortoise on docker there's an optional dependency - _deepspeed_ which hasn't been incorporated into requirements yet, but could be a nice boost.